server/src/main/java/org/elasticsearch/common/geo/builders/PolygonBuilder.java

```diff
@@ -357,7 +357,7 @@ protected static Polygon polygonS4J(GeometryFactory factory, Coordinate[][] poly
         double[] x = new double[shell.length];
         double[] y = new double[shell.length];
         for (int i = 0; i < shell.length; ++i) {
-            x[i] = normalizeLon(shell[i].x);
+            x[i] = Math.abs(shell[i].x) > 180 ? normalizeLon(shell[i].x) : shell[i].x;
```

Comment:
Could you clarify why this is done only here and not in `toPolygonLucene()` and in `LineStringBuilder`?

Reply:
I am not an expert in JTS, but I guess the implementation understand that -180/+180 are the same point and it creates the right polygon. Lucene implementation, and in particular the tesselator implementation does not understand that and it won't create the right polygon. It will join the points without crossing the dateline. 

I don't think it is a bug in Lucene's implementation but a feature and we need to adapt accordingly.